### PR TITLE
PublicDashboards: Fix flaky e2e test

### PR DESCRIPTION
**Pull request by evictorero**
**What this PR does / why we need it**:
We disabled an e2e flaky test a few weeks ago because it was failing randomly. This is an attempt to fix it properly.
- Enabled the test again
- Add a wait to an async call to the backend so the next asserts don't fail

### CODE REVIEW

File: e2e/dashboards-suite/dashboard-public-create.spec.ts [hunk 1]
```diff
@@ -114,7 +114,9 @@ e2e.scenario({
     e2e.pages.ShareDashboardModal.PublicDashboard.EnableSwitch().click({ force: true });
 
     // Save configuration
+    e2e().intercept('/api/dashboards/uid/ZqZnVvFZz/public-config').as('save');
     e2e.pages.ShareDashboardModal.PublicDashboard.SaveConfigButton().click();
+    e2e().wait('@save');
```
Comment (owensmallwood) [inline e2e/dashboards-suite/dashboard-public-create.spec.ts:119]: I've never written an e2e, but is this something we should always do after making a request?

**Review by owensmallwood — APPROVED** [review]
Looks good. Just left a curiosity comment 👍

File: e2e/dashboards-suite/dashboard-public-create.spec.ts (hunk 1 above)
Comment (evictorero) [reply e2e/dashboards-suite/dashboard-public-create.spec.ts:121]: Only when the next assertions could fail because depends on this request result.

**Review by jalevin — APPROVED** [review]
I think I missed code reviewing the original PR for this. Fantastic documentation. Thank you! This looks great.

**Pull request merged**